File: C++/13900_순서쌍의곱의합.cpp
#include <iostream>
using namespace std;

// 2중 루프를 이용해서 계산하면 시간초과
int main() {
    int N;
    cin >> N;
    
    int *num = new int[N];
    long long int sum = 0;
    for(int i = 0; i < N; i++){
        cin >> num[i];
        sum += num[i];
    }
    
    long long int result = 0;
    for(int i = 0; i < N; i++){
        sum -= num[i];
        result += num[i] * sum;
    }
    
    cout << result;
    
    return 0;
}

File: Python/13900_순서쌍의곱의합.py
if __name__ == '__main__':
	N = int(input())

	num = input()
	num = num.split()

	sum = 0
	for i in range(N):
		sum += int(num[i])

	result = 0
	for i in range(N):
		sum -= int(num[i])
		result += int(num[i])*sum

	print(result)
